overlord/snapstate/snapstate.go

```diff
@@ -2688,7 +2696,23 @@ func RemoveMany(st *state.State, names []string) ([]string, []*state.TaskSet, er
 	return removed, tasksets, nil
 }
 
-func validateSnapNames(names []string) error {
+// validateSnapNames returns an error if a snap name
+// (only one condition needs to be met):
+//	* differs from the expected format
+//	* is duplicated in the list
+func validateSnapNames(op string, names []string) error {
```

Comment:
since we're changing it, this should probably be named like `validateSnapNamesForOp(...)` since a generic validation would probably be placed in `snap/naming`, and this one here is operation specific and returns an contextualized error.